frontend/src/app/offers/offer-detail/offer-detail.component.ts

```diff
@@ -1,29 +1,54 @@
-import 'rxjs/add/operator/switchMap';
 import { ActivatedRoute } from '@angular/router';
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy, Inject } from '@angular/core';
+import { DOCUMENT } from '@angular/common';
 import { Observable } from 'rxjs/Observable';
+import { ISubscription } from 'rxjs/Subscription';
+import 'rxjs/add/operator/switchMap';
 
 import { environment } from '../../../environments/environment';
 import { Offer } from '../../homepage-offer/offers.model';
 import { OffersService } from '../../homepage-offer/offers.service';
+import { VolontuloMetaService, VolontuloMetaData } from '../../services/volontulo-meta.service';
 
 @Component({
   selector: 'volontulo-offer-detail',
   templateUrl: './offer-detail.component.html',
   styleUrls: ['./offer-detail.component.scss'],
 })
-export class OfferDetailComponent implements OnInit {
+export class OfferDetailComponent implements OnInit, OnDestroy {
   public offer$: Observable<Offer>;
   public djangoRoot = environment.djangoRoot;
   public getJoinViewUrl = this.offersService.getJoinViewUrl;
+  private metaSubscription: ISubscription;
 
   constructor(
     private activatedRoute: ActivatedRoute,
     private offersService: OffersService,
+    private metaService: VolontuloMetaService,
+    @Inject(DOCUMENT) private document: any
   ) { }
 
   ngOnInit() {
     this.offer$ = this.activatedRoute.params
-    .switchMap(params => this.offersService.getOffer(params.offerId));
+      .switchMap(params => this.offersService.getOffer(params.offerId));
+    this.metaSubscription = this.offer$.subscribe(this.updatePageMetaData.bind(this));
```

Comment:
If we want to explicitly run it here, we don't need this event suscriber on router at all (+ `updatePageMetaData` don't use `this`, so there's no reason to `bind` it).